src/vmm/src/config/builder.rs

```diff
@@ -0,0 +1,354 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Config builder
+use std::convert::TryFrom;
+
+use super::{
+    BlockConfig, ConversionError, KernelConfig, MemoryConfig, NetConfig, VMMConfig, VcpuConfig,
+};
+
+/// Builder structure for VMMConfig
+#[derive(Debug)]
+pub struct Builder {
+    inner: Result<VMMConfig, ConversionError>,
+}
+
+impl Default for Builder {
+    fn default() -> Self {
+        Builder {
+            inner: Ok(VMMConfig::default()),
+        }
+    }
+}
+
+impl Builder {
+    /// Creates a Builder Object
+    ///
+    pub fn new() -> Self {
+        Builder::default()
+    }
+
+    /// Builds VMMConfig
```

Comment:
For consistency reasons, can you please end all sentences in documentation with `.`?